packages/graph/hash_graph/lib/graph/src/store/query/filter.rs

```diff
@@ -189,17 +225,115 @@ impl<'q, T: QueryRecord> TryFrom<Expression> for Option<FilterExpression<'q, T>>
 #[derive(Debug, PartialEq, Deserialize)]
 #[serde(untagged)]
 pub enum Parameter<'q> {
+    Boolean(bool),
     Number(f64),
     Text(Cow<'q, str>),
-    Boolean(bool),
+    #[serde(skip)]
```

Comment:
Can we just deserialize into `Uuid` if we put this near the top of the parameters so serde hits it first?

Reply:
This is potentially more explicit so I'm not suggesting that is better.

Reply:
This leads to an issue that if you have a parameter, which _could_ be converted to `Uuid`, it will get become a `Uuid`, even if a string is expected

Reply:
Gotcha, this is sensible, thanks!